src/utilities/request.jl

```diff
@@ -1,3 +1,23 @@
+# this is used to allow custom dispatches to `_http_request`
+abstract type AbstractBackend end
+
+"""
+    AWS.HTTPBackend <: AWS.AbstractBackend
+
+An `HTTPBackend` can hold default `http_options::AbstractDict{Symbol,<:Any}`
+to pass to HTTP.jl, which can be overwritten per-request by any `http_options`
+supplied there.
+"""
+struct HTTPBackend <: AbstractBackend
+    http_options::AbstractDict{Symbol,<:Any}
+end
+
+function HTTPBackend(; kwargs...)
+    isempty(kwargs) ? HTTPBackend(LittleDict{Symbol, Any}()) : HTTPBackend(LittleDict(kwargs))
+end
+
+const DEFAULT_BACKEND = Ref{AbstractBackend}(HTTPBackend())
```

Comment:
I think we're fine in this case but sometimes to work best with PackageCompiler you want to initialize inside the `__init__`. Just calling this out if you discover some weird PackageCompiler issue with swapping out the backend

Reply:
Ah ok, like re-initialize inside the `__init__`, or add back a `nothing` option and only initialize inside `__init__`?

Reply:
Initializing inside of `__init__` and just leave this undefined:
```suggestion
const DEFAULT_BACKEND = Ref{AbstractBackend}()
```

Reply:
ahh, didn't know one could do that, thanks!